x-pack/plugin/src/test/resources/rest-api-spec/test/ml/get_trained_model_stats.yml

```diff
@@ -3,73 +3,83 @@ setup:
         body: >
           {
-            "model_id": "a-unused-regression-model",
-            "created_by": "ml_tests",
-            "version": "8.0.0",
-            "inference_config": {"regression": {}},
             "description": "empty model for tests",
-            "create_time": 0,
-            "doc_type": "trained_model_config"
+            "tags": ["regression", "tag1"],
+            "input": {"field_names": ["field1", "field2"]},
+            "inference_config": {"regression": {}},
+            "definition": {
+               "preprocessors": [],
+               "trained_model": {
+                  "tree": {
+                     "feature_names": ["field1", "field2"],
+                     "tree_structure": [
+                        {"node_index": 0, "leaf_value": 1}
+                     ],
+                     "target_type": "regression"
+                  }
+               }
+            }
           }
+
   - do:
-      warnings:
-        - "this request accesses system indices: [.ml-inference-000003], but in a future major version, direct access to system indices will be prevented by default"
       headers:
         Authorization: "Basic eF9wYWNrX3Jlc3RfdXNlcjp4LXBhY2stdGVzdC1wYXNzd29yZA==" # run as x_pack_rest_user, i.e. the test setup superuser
-      index:
-        id: trained_model_config-a-used-regression-model-0
-        index: .ml-inference-000003
+      ml.put_trained_model:
+        model_id: a-used-regression-model
         body: >
           {
-            "model_id": "a-used-regression-model",
-            "created_by": "ml_tests",
-            "version": "8.0.0",
-            "inference_config": {"regression": {}},
             "description": "empty model for tests",
-            "create_time": 0,
-            "doc_type": "trained_model_config"
+            "tags": ["regression", "tag1"],
+            "input": {"field_names": ["field1", "field2"]},
+            "inference_config": {"regression": {}},
+            "definition": {
+               "preprocessors": [],
+               "trained_model": {
+                  "tree": {
+                     "feature_names": ["field1", "field2"],
+                     "tree_structure": [
+                        {"node_index": 0, "leaf_value": 1}
+                     ],
+                     "target_type": "regression"
+                  }
+               }
+            }
           }
 
-  - do:
-      warnings:
-        - "this request accesses system indices: [.ml-inference-000003], but in a future major version, direct access to system indices will be prevented by default"
-      headers:
-        Authorization: "Basic eF9wYWNrX3Jlc3RfdXNlcjp4LXBhY2stdGVzdC1wYXNzd29yZA==" # run as x_pack_rest_user, i.e. the test setup superuser
-      indices.refresh:
-        index: ".ml-inference-*"
-
   - do:
       headers:
         Authorization: "Basic eF9wYWNrX3Jlc3RfdXNlcjp4LXBhY2stdGVzdC1wYXNzd29yZA==" # run as x_pack_rest_user, i.e. the test setup superuser
```

Comment:
We should just remove all pipeline creation in these tests and have yaml tests only check on basic crud things. There is still the chance that the state persister makes the stats index because the model is referenced by a pipeline (which gets cleaned up in the "after test" step). 

Consequently the below tests should be changed as well so that ingest stats are not returned.

Reply:
👍  I've removed the pipeline and added those same assertions to `InferenceProcessorIT` so we don't lose any coverage.